### ISSUE TITLE
Undocumented property of extension that implements a protocol gets previous comment in the file

### ISSUE THREAD

**Issue by johnmckerrell** [realm/jazzy#751]
When running jazzy against the following Swift file, the property `shouldBeHidden` on the `Jazzy` struct extension gets documented with "Make some noise". If I remove the documentation comment on the property in the protocol definition then it no longer shows up at all (which I think is the correct thing as the property on the extension is not documented). Interestingly if I add a normal comment, e.g. before the extension line:

```
// Implement the Genre protocol
```

Then `shouldBeHidden` gets blank documentation, but is still documented.

I can live with the property showing up in the documentation, but it definitely doesn't seem right that it's picking up arbitrary comments from further up.

```
import Foundation

public protocol Genre {
    //If the following documentation line is removed, then `shouldBeHidden` doesn't show up in the `Jazzy` extension
    /// This should be hidden in the implementation
    var shouldBeHidden: Bool { get }
}

/// A Jazzy struct ♪♫
public struct Jazzy {
    /// An identifier
    public var id: String
    /// A name
    public var name: String
    
    public var hidden: Bool
    
    /// Make some noise
    public func trumpet() {
        print("Parp!")
    }
}

extension Jazzy: Genre {
    public var shouldBeHidden: Bool {
        return true
    }
}
```

**Comment by jpsim** [comment]
Thanks for filing this, and with enough detail to investigate what's happening. The issue is likely in [SourceKitten](https://github.com/jpsim/SourceKitten) though.

**Comment by johnfairh** [comment]
This is fixed in jazzy 0.8.1 via #815.

Running jazzy against the code above puts the docs in the right place and marks the parts without docs as undocumented.  If you drop the doc comment on `Genre.shouldBeHidden` and use `--skip-undocumented` then all references to `shouldBeHidden` go away.
